File: backend/app/models/municipio_model.py
from sqlalchemy import Column, Float, Integer, String
from app.database import Base


class Municipio(Base):
    __tablename__ = "municipios"

    id            = Column(Integer, primary_key=True, index=True)
    nombre        = Column(String, nullable=False)
    lat           = Column(Float, nullable=False)
    lng           = Column(Float, nullable=False)
    region        = Column(String)          # chiquitania | chaco | norte_integrado | metropolitana | ...
    poblacion     = Column(Integer)
    nivel_riesgo  = Column(String, default="bajo")   # bajo | medio | alto | critico
    score_riesgo  = Column(Float,  default=0.0)      # 0.0 – 1.0

File: frontend/src/app/core/models/municipio.model.ts
export interface Municipio {
  id: number;
  nombre: string;
  lat: number;
  lng: number;
  region: string;
  poblacion: number;
  nivel_riesgo: 'bajo' | 'medio' | 'alto' | 'critico';
  score_riesgo: number;
}

export interface Prediccion {
  tendencia: string;
  meses_criticos: number;
  proyeccion: number[];
  precipitacion_historica: number[];
}

export interface Alerta {
  id: number;
  municipio_id: number;
  municipio_nombre: string;
  tipo: string;
  descripcion: string;
  severidad: 'warning' | 'critical';
  created_at: string;
}

export interface Reading {
  timestamp: string;
  municipio_id: number;
  nivel_freatico_m: number;
  humedad_suelo_pct: number;
  extraccion_lps: number;
  precipitacion_mm: number;
}
